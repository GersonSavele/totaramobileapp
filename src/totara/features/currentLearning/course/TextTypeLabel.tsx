/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import React from "react";
import { Text, View } from "react-native";

import { AppliedTheme } from "@totara/theme/Theme";
import { textStyles } from "@totara/theme/activityLabel";

type ActivityLabelProps = {
  label: any;
  theme: AppliedTheme;
};
const TextTypeLabel = ({ label = {}, theme }: ActivityLabelProps) => {
  const description = validationProperty(label.description);
  return description ? (
    <View style={textStyles.container}>
      {description ? (
        <Text
          style={{
            ...textStyles.labelTextDescription,
            color: theme.colorNeutral6
          }}>
          {description}
        </Text>
      ) : null}
    </View>
  ) : null;
};

const validationProperty = (property: string) => {
  return property && property.trim().length > 0 ? property.trim() : null;
};

export default TextTypeLabel;
