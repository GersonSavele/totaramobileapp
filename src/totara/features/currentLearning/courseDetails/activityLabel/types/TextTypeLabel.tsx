/*

This file is part of Totara Enterprise.
*
Copyright (C) 2020 onwards Totara Learning Solutions LTD
*
Totara Enterprise is provided only to Totara Learning Solutions
LTDâ€™s customers and partners, pursuant to the terms and
conditions of a separate agreement with Totara Learning
Solutions LTD or its affiliate.
*
If you do not have an agreement with Totara Learning Solutions
LTD, you may not access, use, modify, or distribute this software.
Please contact [sales@totaralearning.com] for more information.
*
*/
import React from "react";
import { Text, View } from "react-native";

import { AppliedTheme } from "@totara/theme/Theme";
import { textStyles } from "@totara/theme/activityLabel";

type ActivityLabelProps = {
  label: any;
  theme: AppliedTheme;
};
const TextTypeLabel = ({ label = {}, theme }: ActivityLabelProps) => {
  const name = validationProperty(label.name);
  const description = validationProperty(label.description);
  return name && description ? (
    <View style={textStyles.container}>
      {name ? (
        <Text
          style={{ ...textStyles.labelTextName, color: theme.colorNeutral6 }}
        >
          {name}
        </Text>
      ) : null}
      {description ? (
        <Text
          style={{
            ...textStyles.labelTextDescription,
            color: theme.colorNeutral6,
          }}
        >
          {description}
        </Text>
      ) : null}
    </View>
  ) : null;
};

const validationProperty = (property: string) => {
  return property && property.trim() !== "" ? property.trim() : null;
};

export default TextTypeLabel;
