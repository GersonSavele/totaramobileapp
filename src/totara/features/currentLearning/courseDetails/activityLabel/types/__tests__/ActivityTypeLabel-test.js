/*

This file is part of Totara Enterprise.
*
Copyright (C) 2020 onwards Totara Learning Solutions LTD
*
Totara Enterprise is provided only to Totara Learning Solutions
LTDâ€™s customers and partners, pursuant to the terms and
conditions of a separate agreement with Totara Learning
Solutions LTD or its affiliate.
*
If you do not have an agreement with Totara Learning Solutions
LTD, you may not access, use, modify, or distribute this software.
Please contact [sales@totaralearning.com] for more information.
*
*/

import renderer from "react-test-renderer";
import React from "react";
import TextTypeLabel from "../TextTypeLabel";

const theme = jest.fn();

const mockEmptyLabel = {
  name: undefined,
  description: undefined,
};

const mockLabelWithName = {
  name: "Some of the activities in this course make brief",
  description: undefined,
};

const mockLabelWithDescription = {
  name: undefined,
  description:
    "Some of the activities in this course make brief reference to the Site Administration menu on the left of the screen",
};

const mockLabelWithNameAndDescription = {
  name: "Some of the activities in this course make brief",
  description:
    "Some of the activities in this course make brief reference to the Site Administration menu on the left of the screen",
};

describe("Activity-Label-Type, testing UI with or without label name and description", () => {
  it("Test result : Both name and description with label", () => {
    const component = renderer.create(
      <TextTypeLabel
        theme={theme}
        label={mockLabelWithNameAndDescription}
      ></TextTypeLabel>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("Test result : only the name with label doesn't include description property with label", () => {
    const component = renderer.create(
      <TextTypeLabel theme={theme} label={mockLabelWithName}></TextTypeLabel>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("Test result : only the description with label doesn't include name property with label", () => {
    const component = renderer.create(
      <TextTypeLabel
        theme={theme}
        label={mockLabelWithDescription}
      ></TextTypeLabel>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("Test result : label doesn't include name property and description with label", () => {
    const component = renderer.create(
      <TextTypeLabel theme={theme} label={mockEmptyLabel}></TextTypeLabel>
    );
    const tree = component.toJSON();
    expect(tree).toBeNull();
  });
});
