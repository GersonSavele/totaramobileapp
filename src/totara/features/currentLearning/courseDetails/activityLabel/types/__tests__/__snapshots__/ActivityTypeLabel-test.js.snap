// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity-Label-Type, testing UI with or without label name and description Test result : Both name and description with label 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
      "margin": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": undefined,
        "fontSize": 31,
        "fontWeight": "bold",
        "marginBottom": 8,
        "textAlign": "left",
      }
    }
  >
    Some of the activities in this course make brief
  </Text>
  <Text
    style={
      Object {
        "color": undefined,
        "fontSize": 31,
        "lineHeight": 40,
        "textAlign": "left",
      }
    }
  >
    Some of the activities in this course make brief reference to the Site Administration menu on the left of the screen
  </Text>
</View>
`;

exports[`Activity-Label-Type, testing UI with or without label name and description Test result : only the description with label doesn't include name property with label 1`] = `null`;

exports[`Activity-Label-Type, testing UI with or without label name and description Test result : only the name with label doesn't include description property with label 1`] = `null`;
